src/librawspeed/decoders/DngDecoderSlices.cpp

```diff
@@ -418,10 +419,9 @@ void DngDecoderSlices::decodeSlice(DngDecoderThread* t) {
     while (!t->slices.empty()) {
       DngSliceElement e = t->slices.front();
       t->slices.pop();
-      LJpegPlain l(*mFile, e.byteOffset, e.byteCount, mRaw);
-      l.mDNGCompatible = mFixLjpeg;
+      DngDecompressor d(*mFile, e.byteOffset, e.byteCount, mRaw);
```

Comment:
> Introducing the term 'Abstract' would certainly be possible <...>. Maybe you want to generally introduce this kind naming hint across the library.

Yeah, it's a flat unstructured mess currently

> but it is also not used in other places, like `RawDecoder`

that is of little concern. the directories weren't used either